File: OLD Repos/Linear Algebra/Linear Operations Foundation/Linear Combination.jl
#### Linear Combination

using LinearAlgebra

function lincomb(coeff, vectors)
    n = length(vectors[1])
    r = zeros(n)
    for i = 1:length(vectors)
        r = r + coeff[i] * vectors[i]
    end
    return r
end

re = lincomb(([-0.5, 1.5]), ([1, 2], [3, 4]))

println(re)     # [4.0, 5.0]

File: Appendix Algebra/Linear Operations Foundation/Linear Combination.py
#### Linear Combination

import numpy as np

def lincomb(coeff, vectors):
    n = vectors[1].size
    r = np.zeros(n)
    for i in range(0,len(vectors)):
        r = r + coeff[i] * vectors[i]
    return r

re = lincomb((np.array([-0.5, 1.5])), (np.array([1, 2]), np.array([3, 4])))

print(re)       # [4. 5.]